src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Indexer/ProductIndexer.php

```diff
@@ -21,92 +20,113 @@
  * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
-class ProductIndexer implements IndexerInterface, BulkIndexerInterface, RemoverInterface, BulkRemoverInterface
+class ProductIndexer implements ProductIndexerInterface
 {
     private const PRODUCT_IDENTIFIER_PREFIX = 'product_';
+
+    /**
+     * Index type is not used anymore in elasticsearch 6, but is still needed by Client
+     */
+    const INDEX_TYPE = 'pim_catalog_product';
+
     /** @var NormalizerInterface */
     private $normalizer;
 
     /** @var Client */
     private $productAndProductModelClient;
 
-    /** @var string */
-    private $indexType;
+    /** @var ProductRepositoryInterface */
+    private $productRepository;
 
     /**
-     * @param NormalizerInterface $normalizer
-     * @param Client              $productAndProductModelClient
-     * @param string              $indexType
+     * @param NormalizerInterface        $normalizer
+     * @param Client                     $productAndProductModelClient
+     * @param ProductRepositoryInterface $productRepository
      */
     public function __construct(
         NormalizerInterface $normalizer,
         Client $productAndProductModelClient,
-        string $indexType
+        ProductRepositoryInterface $productRepository
     ) {
         $this->normalizer = $normalizer;
         $this->productAndProductModelClient = $productAndProductModelClient;
-        $this->indexType = $indexType;
+        $this->productRepository = $productRepository;
     }
 
     /**
      * Indexes a product in both the product index and the product and product model index.
      *
      * {@inheritdoc}
      */
-    public function index($object, array $options = []) : void
+    public function indexFromProductIdentifier(string $productIdentifier, array $options = []): void
     {
+        $object = $this->productRepository->findOneByIdentifier($productIdentifier);
+        if (!$object instanceof ProductInterface) {
+            return;
+        }
+
         $normalizedObject = $this->normalizer->normalize(
             $object,
             ProductModelNormalizer::INDEXING_FORMAT_PRODUCT_AND_MODEL_INDEX
         );
         $this->validateObjectNormalization($normalizedObject);
-        $this->productAndProductModelClient->index($this->indexType, $normalizedObject['id'], $normalizedObject);
+        $this->productAndProductModelClient->index(self::INDEX_TYPE, $normalizedObject['id'], $normalizedObject);
     }
 
     /**
-     * Indexes a product in both the product index and the product and product model index.
+     * Indexes a list of products in both the product index and the product and product model index.
```

Comment:
```suggestion
     * Indexes a list of products in the product and product model index from their identifiers .
```

(there is only one index left); please update the comment for method `indexFromProductIdentifier()` as well